### PR TITLE
[clang][Driver] Allow -fuse-lld=lld-link when lto is enabled on *windows-msvc targets

### PR DESCRIPTION
**Pull request by zhaoshiz**
Follow-up on https://github.com/llvm/llvm-project/pull/109607, we have a use case on WoA where `cmake -G "Unix Makefiles"` generates -fuse-ld=lld-link, which is disallowed by PR#109607.

### CODE REVIEW

File: clang/test/Driver/woa-lto.c
Comment (MaxEW707) [reply clang/test/Driver/woa-lto.c:15]: On second thought, `{{(.exe)?}}` is better suffix check.
So `"{{.*}}lld-link{{(.exe)?}}"`.

**Comment by MaskRay** [comment]
Windows should use `-fuse-ld=lld` as well, not `-fuse-ld=lld-link`

**Comment by MaxEW707** [comment]
> Windows should use `-fuse-ld=lld` as well, not `-fuse-ld=lld-link`

Is `-fuse-ld=lld-link` intended to be an allowable option?

Right now we don't issue any error if a user passes in `lld-link`.

I am interpreting your statement as suggesting we should do something like following?
```
const Arg *A = Args.getLastArg(options::OPT_fuse_ld_EQ);
if (A->getValue() == "lld-link") {
    D.Diag(diag::err_drv_unsupported_option_argument) << A->getSpelling() << A->getValue();
}
```

**Comment by zhaoshiz** [comment]
> Windows should use `-fuse-ld=lld` as well, not `-fuse-ld=lld-link`

I don't disagree... But `-fuse-ld=lld-link` is auto generated by CMake in our case. I can add `-fuse-ld=lld` to CMAKE_C_FLAGS. But I have no control over what or how our downstream users build with LLVM.

**Comment by MaxEW707** [comment]
> > Windows should use `-fuse-ld=lld` as well, not `-fuse-ld=lld-link`
> 
> I don't disagree... But `-fuse-ld=lld-link` is auto generated by CMake in our case. I can add `-fuse-ld=lld` to CMAKE_C_FLAGS. But I have no control over what or how our downstream users build with LLVM.

That is a fair point. CMake docs even site using `-fuse-ld=lld-link` in their examples.
https://github.com/Kitware/CMake/blob/ae6561699dd22226b7d079ec946e8032bcf0a6e4/Help/variable/CMAKE_LANG_USING_LINKER_TYPE.rst#L28

We might just have to continue supporting `lld-link` as it worked before.
I'll let @MaskRay be the final say in what direction to go. Otherwise the PR looks good to me :).

File: clang/test/Driver/woa-lto.c
Comment (MaskRay) [inline clang/test/Driver/woa-lto.c:1]: "woa-" is not a clear filename prefix. Suggest `windows-lto.c`

In general, only -### should be used in driver tests to suppress action running. We can then drop `REQUIRES: `.

File: clang/test/Driver/woa-lto.c
Comment (MaskRay) [inline clang/test/Driver/woa-lto.c:3]: Just use `%s` and avoid this temporary .c file.

**Comment by zhaoshiz** [comment]
Thanks @MaskRay, I've revised and renamed the test file.

**Review by MaxEW707 — APPROVED** [review]
LGTM!

Would like a final look over from @MaskRay before merging :).

**Review by MaskRay — APPROVED** [review]

File: clang/test/Driver/windows-lto.c
Comment (MaskRay) [inline clang/test/Driver/windows-lto.c:3]: delete `^//$` lines

**Pull request merged**
